Model/Order/PlaceOrder.php

```diff
@@ -261,158 +256,41 @@ public function authorizeAndPlace(string $publicOrderId, string $quoteMaskId): R
             return $this->getValidationErrorResponse($e->getMessage());
         }
 
-        if ($quote->getId() === null) {
```

Comment:
It seems like this check for null quote id was removed, is this intentional?

Reply:
Great question! 
If the quote doesn't have an ID it will be caught [here if it's masked](https://github.com/bold-commerce/adobe-commerce-bold-checkout/pull/303/files#diff-ffeb6436be545c3a792a30e5098602eca8373f7f95755112c5fe04afb1835a51L259) or [here](https://github.com/bold-commerce/adobe-commerce-bold-checkout/pull/303/files#diff-ffeb6436be545c3a792a30e5098602eca8373f7f95755112c5fe04afb1835a51L261) if it's not. Essentially in the two methods above. So from my understanding this check it's doing anything, but please always sanity check me.

Reply:
At the time that check was added, there was no verification in the logic that was retrieving the quote that it actually existed. If it didn't, an empty quote object would be returned. That bug was fixed in a later commit to the retrieval logic by Pavel, rendering this work-around no longer necessary.